React/Views/RCTView.m

```diff
@@ -184,6 +184,54 @@ - (BOOL)didActivateAccessibilityCustomAction:(UIAccessibilityCustomAction *)acti
   return YES;
 }
 
+- (NSString *)accessibilityValue
+{
+  NSMutableString *value = [NSMutableString stringWithString:@""];
+  NSDictionary *roleDescriptions = @{
+                                     @"alert" : @"alert",
+                                     @"checkbox" : @"checkbox",
+                                     @"combobox" : @"combo box",
+                                     @"editabletext" : @"edit",
+                                     @"menu" : @"menu",
+                                     @"menubar" : @"menu bar",
+                                     @"menuitem" : @"menu item",
+                                     @"progressbar" : @"progress bar",
+                                     @"radiobutton" : @"radio button",
+                                     @"radiogroup" : @"radio group",
+                                     @"scrollbar" : @"scroll bar",
+                                     @"spinbutton" : @"spin button",
+                                     @"switch" : @"switch",
+                                     @"tab" : @"tab",
+                                     @"tablist" : @"tab list",
+                                     @"timer" : @"timer",
+                                     @"toolbar" : @"tool bar",
+                                     };
+  NSDictionary *stateDescriptions = @{
+                                      @"on" : @"on",
+                                      @"off" : @"off",
+                                      @"checked" : @"checked",
+                                      @"unchecked" : @"not checked",
+                                      @"busy" : @"busy",
+                                      @"expanded" : @"expanded",
+                                      @"collapsed" : @"collapsed",
+                                      };
+  NSString *roleDescription = _accessibilityRole ? roleDescriptions[_accessibilityRole]: nil;
+  if (roleDescription) {
+    [value appendString:roleDescription];
+  }
+  for (NSString *state in _accessibilityStates) {
+    NSString *stateDescription =state ? stateDescriptions[state] : nil;
+    if (stateDescription) {
+        [value appendString:@" "];
+    [value appendString:stateDescription];
```

Comment:
Like you called out in your email, this doesn't localize well. We'll need to figure something out going forward.

Reply:
Any thoughts on how to do this? Since React Native is built as a static library, it wasn't clear to me how to ensure that string resources get bundled with apps that use it. This is already handled in Android.

Reply:
The classic iOS way is to use an `NSLocalizedString` here with translations in a `Localizable.strings` file in the bundle. Delivering the english only version of Localizable.strings seems reasonable, perhaps we can do that via CocoaPods?

Reply:
This sounds promising. Any pointers at how to do this?

also, I'd like to use RNTester to validate this approach, and I don't think it's using CoCoPods yet, is it?

Reply:
This article explains `NSLocalizedString` pretty well https://nshipster.com/nslocalizedstring/

I'm not a CocoaPods expert, so that part I'm shaky on, perhaps ask in RN Discord (let me know if you need a link). If RNTester isn't using CocoaPods, it should be pretty easy to get translations working there. The trickier park will be integrating into apps that might already have a Localizable.strings file. 

Regardless, I think the localization bit should be a separate PR :)

Reply:
Yep, I understand how to add localizable strings to an iOS project. We actually did that in some earlier internal revisions. The problem was getting those localizable strings into dependent projects, since React Native is built as a static library.

Unfortunately, there's no way to remove this code and still have the fallback paths functional. Do you propose we move forward with this PR as written, and then create a separate one to add the localized strings? If so, we'll need some support from Facebook to figure out a way to ensure that React Native apps get the localizable strings by default.

Reply:
A feasible way to accomplish localizable roles and states would be having new props just override the default descriptions. 

The API would be like this:

```js
// src/index.js

// change descriptions globally from my app
if (currentLocale != 'en') {
  View.defaultProps.accessibilityRoleDescriptions = {
    radiogroup: translate('accessibility.role.radiogroup'),
  }

  View.defaultProps.accessibilityStateDescriptions = {
    expanded: translate('accessibility.states.radiogroup'),
  }
}
```

`accessibilityRoleDescriptions` and `accessibilityStateDescriptions` would only override prop keys passed from `stateDescriptions` and `roleDescriptions` arrays, for both iOS and Android.

Just to point out, IMO we could merge this PR without this localizable solution in case it demands too much. I'd rather have a non-localizable solution now than have nothing. And if case some help is needed, to have it in a follow-up PR, let me know.

Reply:
Something like this could work, although it requires every app to maintain their own role and state descriptions. I'd love some mechanism that deploys localized descriptions into the new app template, so app developers don't have to think about it.

I'm in favor of moving forward with this PR as is, and creating an additional PR to sort out localization of the role and state descriptions.